query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -189,6 +358,46 @@ public void handleRequest(
     }
   }
 
+  @Nonnull
+  private QueryRequest removeViewColumnFilters(QueryRequest request,
+      Map<String, ViewColumnFilter> columnFilterMap) {
+
+    Filter filter = request.getFilter();
+    for (String columnName: columnFilterMap.keySet()) {
+      filter = removeViewColumnFilter(filter, columnName);
+      if (filter == null) {
+        break;
+      }
+    }
+
+    QueryRequest.Builder builder = QueryRequest.newBuilder(request);
+    if (filter == null) {
+      return builder.clearFilter().build();
+    } else {
+      return builder.clearFilter().setFilter(filter).build();
+    }
+  }
+
+  @Nullable
+  private Filter removeViewColumnFilter(Filter filter, String columnName) {
+    if (filter.getChildFilterCount() > 0) {
+      // Recursively try to remove the filter and eliminate the null nodes.
+      List<Filter> newFilters = filter.getChildFilterList().stream()
+          .map(f -> removeViewColumnFilter(f, columnName))
+          .filter(Objects::nonNull).collect(Collectors.toList());
+      if (newFilters.size() == 1) {
+        return newFilters.get(0);
+      } else {
+        return Filter.newBuilder(filter).clearChildFilter().addAllChildFilter(newFilters).build();
+      }
+    } else {
+      if (filter.getLhs().getColumnIdentifier().getColumnName().equals(columnName)) {
```

Comment:
What if columnIdentifier is not set? Can that even happen with proto?

Reply:
ideally not but I can add a requirement here so that we don't run into NPE.

Reply:
It can't - protos are never null!
```java
  public org.hypertrace.core.query.service.api.ColumnIdentifier getColumnIdentifier() {
    if (valueCase_ == 1) {
       return (org.hypertrace.core.query.service.api.ColumnIdentifier) value_;
    }
    return org.hypertrace.core.query.service.api.ColumnIdentifier.getDefaultInstance();
  }
```

Reply:
Done.